packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -190,6 +190,15 @@ actual object RealmInterop {
         realmc.realm_close((realm as LongPointerWrapper).ptr)
     }
 
+    actual fun realm_delete_files(path: String) {
+        val deleted = booleanArrayOf(false)
+        realmc.realm_delete_files(path, deleted)
+
+        if (!deleted[0]) {
```

Comment:
Seems like the implementation differs on the platforms. If we throw on `!deleted[0]` we should probably also checked `deleted` on Kotlin Native!